bitnami/kafka/values.yaml

```diff
@@ -342,6 +342,39 @@ auth:
     ## ref: https://docs.confluent.io/current/kafka/authentication_ssl.html#optional-settings
     ##
     endpointIdentificationAlgorithm: https
+  ## Zookeeper client configuration for kafka brokers
+  ##
+  zookeeper:
+    ## TLS configuration
+    ##
+    tls:
+      ## @param auth.zookeeper.tls.enabled Enable TLS for Zookeeper client connections.
+      ##
+      enabled: false
+      ## @param auth.zookeeper.tls.type Format to use for TLS certificates. Allowed types: `jks` and `pem`.
+      ##
+      type: jks
+      ## @param auth.zookeeper.tls.verifyHostname Hostname validation.
+      ##
+      verifyHostname: true
+      ## @param auth.zookeeper.tls.existingSecret Name of the existing secret containing the TLS certificates for ZooKeeper client communications.
+      ##
+      existingSecret: ""
```

Comment:
After taking another look at the changes they LGTM overall, good job! However, I do have a small concern here.

I am not a Kafka expert, so maybe you have more background on this than me. Looking at the chart, it seems that in order to configure TLS for Kafka itself, one has to configure a TLS certificate for EACH broker. I have also found this in the documentation:

https://kafka.apache.org/documentation/#security_ssl_key

Because of that, the Bitnami Kafka chart exposes an **array** of `existingSecrets`. However, they are not explicit when it comes to Zookeeper TLS:

https://kafka.apache.org/documentation/#zk_authz

> When using mTLS alone, every broker and any CLI tools (such as the [ZooKeeper Security Migration Tool](https://kafka.apache.org/documentation/#zk_authz_migration)) should identify itself with the same Distinguished Name (DN) because it is the DN that is ACL'ed

This leaves me in doubt here on whether multiple certs should be used for Zookeeper auth as well or not. Technically, it should be possible to use the same client certs as they mention that is the DN that is ACL, so maybe using a TLS per broker should be regarded as an improvement of this PR.

What are your thoughts on this? Have you tried these changes with more than one broker?

Reply:
Hi @joancafom!
Thank you!

> What are your thoughts on this? Have you tried these changes with more than one broker?
Yes, I tried these changes with 3 Kafka brokers and 3 Zookeeper nodes. If you use the same certificate for Zookeeper auth, you must have the same DN and add all possible SAN to allow hostname validation. Using the same DN is a requirement if you use mtls auth only (unless you write a custom auth provider). If you use SASL+mtls, then you can configure the DN as you want since they will use the ACL'ed SASL. 

To be honest, it can be interesting to add the ability to configure one TLS certificate used for Zookeeper auth per broker. However, I would implement this later as an improvement of this PR :). This PR already provides the ability to configure mtls with Zookeeper et does not restrict the user possibilities.

Reply:
Agreed, thank you so much 🚀